packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/computeAxisValue.ts

```diff
@@ -106,138 +102,118 @@ export function computeAxisValue<T extends ChartSeriesType>({
+          triggerTooltip,
+          ...axis,
+          data,
+          scale,
+          tickNumber: axis.data!.length,
+          colorScale:
+            axis.colorMap &&
+            (axis.colorMap.type === 'ordinal'
+              ? getOrdinalColorScale({ values: axis.data, ...axis.colorMap })
+              : getColorScale(axis.colorMap)),
+        };
       }
-    }
-    if (isPointScaleConfig(axis)) {
-      const scaleRange = axisDirection === 'y' ? [...range].reverse() : range;
-      const zoomedRange = zoomScaleRange(scaleRange, zoomRange);
-
-      completeAxis[axis.id] = {
-        offset: 0,
-        height: 0,
-        triggerTooltip,
-        ...axis,
-        data,
-        scale: scalePoint(axis.data!, zoomedRange),
-        tickNumber: axis.data!.length,
-        colorScale:
-          axis.colorMap &&
-          (axis.colorMap.type === 'ordinal'
-            ? getOrdinalColorScale({ values: axis.data, ...axis.colorMap })
-            : getColorScale(axis.colorMap)),
-      };
 
       if (isDateData(axis.data)) {
         const dateFormatter = createDateFormatter(axis.data, scaleRange, axis.tickNumber);
         completeAxis[axis.id].valueFormatter = axis.valueFormatter ?? dateFormatter;
       }
+
+      return;
     }
 
     if (axis.scaleType === 'band' || axis.scaleType === 'point') {
       // Could be merged with the two previous "if conditions" but then TS does not get that `axis.scaleType` can't be `band` or `point`.
       return;
     }
 
-    const scaleType = axis.scaleType ?? ('linear' as const);
-
-    const domainLimit = preferStrictDomainInLineCharts
-      ? getAxisDomainLimit(axis, axisDirection, axisIndex, formattedSeries)
-      : (axis.domainLimit ?? 'nice');
-
-    const axisExtremums = [axis.min ?? minData, axis.max ?? maxData];
-
-    if (typeof domainLimit === 'function') {
-      const { min, max } = domainLimit(minData, maxData);
-      axisExtremums[0] = min;
-      axisExtremums[1] = max;
-    }
-
-    const rawTickNumber = getTickNumber({ ...axis, range, domain: axisExtremums });
+    const rawTickNumber = scaleDefinition.tickNumber!;
+    const continuousAxis = axis as Readonly<
+      DefaultedAxis<ContinuousScaleName, any, Readonly<ChartsAxisProps>>
+    >;
+    const scaleType = continuousAxis.scaleType ?? ('linear' as const);
     const tickNumber = scaleTickNumberByRange(rawTickNumber, zoomRange);
 
-    const zoomedRange = zoomScaleRange(range, zoomRange);
-
-    const scale = getScale(scaleType, axisExtremums, zoomedRange);
-
-    if (isSymlogScaleConfig(axis) && axis.constant != null) {
-      (scale as ScaleSymLog<number, number>).constant(axis.constant);
+    if (filter) {
+      const [minData, maxData] = getAxisExtremum(
+        axis,
+        axisDirection,
+        seriesConfig as ChartSeriesConfig<CartesianChartSeriesType>,
+        axisIndex,
+        formattedSeries,
+        filter,
+      );
+      scale = scale.copy();
+      scale.domain([minData, maxData]);
```

Comment:
This should be slightly more complex. The filters is here to remove some data points.

But it should still apply domainLimit strategies: `nice(tickNumber)` or the `domainLimit(min, max)` if defined.


We could add a test with a BarChart with series data set to `[1, 4.5, 5]` and x-axis zoom set to `[0, 50]` (ditch the last item)

- `domainLimit='nice'`: y-axis should go from 0 to 5
- `domainLimit='strict'`: y-axis should go from 0 to 4.5

Dark mode is the actual docs demo, light mode is this PR preview


https://github.com/user-attachments/assets/ef2fa7e7-38ae-4e4c-85de-a38797384b03

This will have a side effect impact: The `domainLimit` as a function could be called twice:
- one when creating scales
- one when applying the filter

Not a big deal for me

Reply:
Should be fixed with the most recent commit 👍